modules/apps/commerce/commerce-product-type-virtual-api/src/main/java/com/liferay/commerce/product/type/virtual/exception/CPDefinitionVirtualSettingWrongCPTypeException.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.product.type.virtual.exception;
+
+import com.liferay.portal.kernel.exception.PortalException;
+
+/**
+ * @author Stefano Motta
+ */
+public class CPDefinitionVirtualSettingWrongCPTypeException
```

Comment:
you should add this exception to the service.xml of commerce-product-type-virtual-service module and run service builder to generate it